juju/client/connection.py

```diff
@@ -871,7 +876,7 @@ async def _connect_with_login(self, endpoints):
             # a few times.
             for i in range(0, 2):
                 result = (await self.login())['response']
-                macaroonJSON = result.get('discharge-required')
+                macaroonJSON = result.get('bakery-discharge-required')
```

Comment:
For the casual observer - newer Juju controllers (2.9 onwards) use the "bakery-discharge-required" attribute.